pycloudlib/lxd/instance.py

```diff
@@ -77,10 +77,14 @@ def parse_ip(self, query: dict):
         network = query.get("state", {}).get("network")
         if network is None:
             network = {}
-        for _, nic_cfg in network.items():
-            if not nic_cfg.get("host_name"):
-                continue
+        for nic_cfg in sorted(
+            network.values(),
+            # prefer nics with host_name defined
+            key=lambda cfg: not cfg.get("host_name"),
+        ):
             for addr in nic_cfg["addresses"]:
+                if addr.get("scope") != "global":
```

Comment:
Do you think, @holmanb, there is a better way to discover IPs in this (see the new test) situations?

Reply:
I don't have anything better at the moment. That looks good to me.